tests/sles4sap/monitoring_services.pm

```diff
@@ -0,0 +1,156 @@
+            systemctl "status prometheus-hanadb_exporter\@$rsc_id";
+        }
+
+        # Check that the exporter is running
+        assert_script_run "curl http://localhost:$hanadb_exporter_port" if $check_exporter eq 'true';
+    }
+
+    # Configure SAP Host Exporter
+    if (get_var('NW')) {
+        my $exporter_conf_name = 'default';
+        my $host_exporter      = 'sap_host_exporter';
+        my $exporter_config    = "/etc/$host_exporter/$exporter_conf_name.yaml";
+        my $check_exporter     = 'false';
+        my $exporter_port;
+
+        # Install needed packages
+        zypper_call "in prometheus-$host_exporter";
+
+        if (get_var('HA_CLUSTER')) {
+            my $exporter_rsc = "rsc_exporter_$rsc_id";
+            $exporter_port  = 9680;
+            $check_exporter = 'true';
+
+            # Use mutex to be sure that only *one* node at a time can access the CIB
+            # NOTE: using 'support_server_ready' mutex because it already exists
+            #       and because mutex should be create at the beginning on support_server
+            mutex_lock 'support_server_ready';
+
+            # Add the monitoring resource
+            assert_script_run(
+                'crm configure primitive ' . $exporter_rsc . ' systemd:prometheus-sap_host_exporter@' . $rsc_id
+                  . ' op start interval=0 timeout=100'
+                  . ' op stop interval=0 timeout=100'
+                  . ' op monitor interval=10'
+                  . ' meta target-role=Stopped'
+            );
+            assert_script_run "crm configure modgroup grp_$rsc_id add $exporter_rsc";
+            assert_script_run "crm resource start $exporter_rsc";
+            wait_until_resources_started;
+
+            # Release the lock
+            mutex_unlock 'support_server_ready';
+        } else {
+            # Get the IP port and start exporter
+            ($exporter_port) = script_output("awk '/port:/ { print \$NF }' $exporter_config", proceed_on_failure => 1) =~ /(\d+)/;
+            $exporter_port //= 9680;
+            $check_exporter = 'true';
+            systemctl "enable --now prometheus-$host_exporter\@$exporter_conf_name";
+            systemctl "status prometheus-$host_exporter\@$exporter_conf_name";
+        }
+
+        # Check that the exporter is running
+        assert_script_run "curl http://localhost:$exporter_port" if $check_exporter eq 'true';
+    }
+
+    # Synchronize the nodes if needed - this is to prevent weird behaviour if a node is stopped before another
+    barrier_wait "MONITORING_CONF_DONE_$cluster_name" if get_var('HA_CLUSTER');
+
+    # Install and start node_exporter
+    zypper_call 'in golang-github-prometheus-node_exporter';
+    systemctl 'enable --now prometheus-node_exporter';
+    systemctl 'status prometheus-node_exporter';
+
+    # Check that node_exporter is working as expected
+    my $monitoring_port //= 9100;
+    assert_script_run "curl http://localhost:$monitoring_port/metrics";
```

Comment:
It could be useful to save the metrics to a file and upload it to the test.

Reply:
Yes, I will add this.

Reply:
Done.